src/utils/helpers.tsx

```diff
@@ -0,0 +1,84 @@
+import { createTodos, deleteTodos } from '../api/todos';
+import { Todo } from '../types/Todo';
+import { errorMessages } from './const';
+import { client } from './fetchClient';
+
+export enum Status {
+  all = 'All',
+  active = 'active',
+  completed = 'completed',
+}
+
+export const filter = (todos: Todo[], selectedFilter: Status): Todo[] => {
+  switch (selectedFilter) {
+    case Status.all:
+      return todos;
+
+    case Status.active:
+      return todos.filter(todo => !todo.completed);
+
+    case Status.completed:
+      return todos.filter(todo => todo.completed);
+
+    default:
+      return todos;
+  }
+};
+
+export const deleteTodo = (
+  todoId: number,
+  setTodos: (update: (todos: Todo[]) => Todo[]) => void,
+  setIsLoadingWhileDelete: (isLoading: boolean) => void,
+  setHasError: (value: boolean) => void,
+  setErrorMessage: (message: string) => void,
+  onSuccess?: () => void,
+) => {
+  setIsLoadingWhileDelete(true);
+
+  deleteTodos(todoId)
+    .then(() => {
+      setTodos(currentTodos => currentTodos.filter(todo => todo.id !== todoId));
+      if (onSuccess) {
+        onSuccess();
+      }
+    })
+    .catch(() => {
+      if (setErrorMessage) {
```

Comment:
- This condition doesn't make sense as it is always true.
- And you actually don't need to move these functions to helpers as you pass so many set state functions, keep them in the components. They are just regular handler functions